c++/src/Reader.cc

```diff
@@ -530,6 +531,9 @@ namespace orc {
           (*indexStats)[column].push_back(entry.statistics());
         }
       }
+      if (streamKind != StreamKind::StreamKind_ROW_INDEX) {
```

Comment:
Can you make this an else?

Reply:
sure.
is WriterImpl::writeStripe() works correctly? probably bloom filter streams should be added to the indexLength, not dataLength.

Reply:
You are right. WriterImpl::writeStripe() should be fixed to include bloom filter index.